src/os_linux.cpp

```diff
@@ -430,4 +432,218 @@ bool OS::checkPreloaded() {
+    FILE* file = fopen(path, "rb");
+    if (!file) return false;
+
+    const size_t max_read = sizeof(info->_cmdline) - 1;
+    size_t len = fread(info->_cmdline, 1, max_read, file);
+    info->_cmdline[max_read] = '\0';
+
+    if (len > 0) {
+      // Replace null bytes with spaces (arguments are separated by null bytes)
+      for (size_t i = 0; i < len; i++) {
+          if (info->_cmdline[i] == '\0') {
+              info->_cmdline[i] = ' ';
+        }
+      }
+      // Ensure null termination
+      info->_cmdline[len] = '\0';
+      // Remove trailing space if present
+      if (len > 0 && info->_cmdline[len-1] == ' ') {
+          info->_cmdline[len-1] = '\0';
+      }
+    }
+
+    fclose(file);
+    return true;
+}
+
+// Helper method to read process stats from /proc/{pid}/stat
+static bool readProcessStats(int pid, ProcessInfo* info) {
+    char path[64];
+    snprintf(path, sizeof(path), "/proc/%d/stat", pid);
+    FILE* file = fopen(path, "r");
+    if (!file) return false;
+
+    char buffer[1024];
+    if (!fgets(buffer, sizeof(buffer), file)) {
+        fclose(file);
+        return false;
+    }
+    fclose(file);
+
+    int  parsed_pid, parsed_ppid;
+    char comm[COMM_LEN] = {0};
+    char state;
+    unsigned long minflt, majflt, utime, stime;
+    unsigned long long starttime64;
+    unsigned long vsize;
+    long threads;
+    long rss;
+
+    int parsed = sscanf(buffer,
+                        "%d "                     /*  1 pid                                   */
+                        "(%15[^)]) "              /*  2 comm  (read until ')')                */
+                        "%c %d "                  /*  3 state, 4 ppid                         */
+                        "%*d %*d %*d %*d %*u "    /*  5-9(skip) pgrp,session,tty,tpgid,flags  */
+                        "%lu %*u %lu %*u "        /*  10-13 minflt,-,majflt,-                 */
+                        "%lu %lu "                /*  14-15 utime, stime                      */
+                        "%*d %*d %*d %*d "        /*  16-19(skip) cutime,cstime,prio,nice     */
+                        "%ld "                    /*  20 num_threads                          */
+                        "%*d "                    /*  21 itrealvalue (skip)                   */
+                        "%llu "                   /*  22 starttime                            */
+                        "%lu "                    /*  23 vsize                                */
+                        "%ld",                    /*  24 rss                                  */
+        &parsed_pid, comm, &state, &parsed_ppid,
+        &minflt, &majflt, &utime, &stime,
+        &threads, &starttime64, &vsize, &rss);
+
+    if (parsed < 12) return false;
+
+    info->_pid           = parsed_pid;
+    info->_ppid          = parsed_ppid;
+    strncpy(info->_name, comm, sizeof(info->_name));
+    info->_name[sizeof(info->_name) - 1] = '\0';
```

Comment:
Usage of `strncpy` looks suspicious to me. Why not just `memcpy`? `comm` is always NUL-terminated.